File: My Answer/E7.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Scanner;

public class E7 {
    public static void Q1(){
        for(int i = 0; i < 10; i++){
            System.out.printf("O treinamento de java começa em %d minutos!\n", (i+1));
        }
    }

    public static void Q2(){
        List<Integer> vendas = new ArrayList<>(Arrays.asList(2000, 5000, 1000, 8000, 3000));
        double comissao = 0.0;
        for(int i = 0; i < vendas.size(); i++){
            double temp = (double)vendas.get(i);
            comissao += (temp > 4000.00) ? temp*0.1:temp*0.05;   
        }
        System.out.printf("Valor total da comissão: %.2f\n", comissao);


    }

    public static void Q3(){
        List<String> estoqueProdutos = new ArrayList<>(Arrays.asList("monitor", "teclado0", "mouse", "headset", "gabinete"));
        List<Integer> quantidadeProdutos = new ArrayList<>(Arrays.asList(5, 12, 2, 8, 15));
        if(estoqueProdutos.size() == quantidadeProdutos.size()){
            for(int i = 0; i < estoqueProdutos.size(); i++){
                int temp = quantidadeProdutos.get(i);
                if(temp < 8) System.out.printf("Alerta: o produto %s está com apenas %d unidades no estoque.\n", estoqueProdutos.get(i), temp);
            }
        }
    }

    public static void Q4(){
        HashMap<String, Integer> metas = new HashMap<>(Map.of(
            "jan", 1000,
            "fev", 1200,
            "mar", 1100
        ));
        HashMap<String, Integer> gastos = new HashMap<>(Map.of(
            "jan", 900,
            "fev", 1350,
            "mar", 1100
        ));
        if(metas.size() == gastos.size()){
            for(String key : metas.keySet()){
                double temp = (double)gastos.get(key);
                double temp2 = metas.get(key);
                if(temp <= temp2) System.out.printf("Mês %s: dentro do orçamento.\n", key);
                else System.out.printf("Mes %s: Orçamento estourado em R$%.2f.\n", key, (temp - temp2));
            }
        }
    }

    public final static Scanner scanner_input = new Scanner(System.in);

    public static void Q5(){
        HashMap<String, Double> precos = new HashMap<>(Map.of(
            "celular", (double)1500,
            "tablet", (double)2500,
            "notebook", (double)5000
        ));
        System.out.print("Digite o percentual de almento: ");
        String percent = scanner_input.nextLine();
        percent = percent.replace("%", "").replace(" ","").replace(",", ".");
        double factor = Double.parseDouble(percent)/100.00;
        System.out.println(factor);
        for(String key:precos.keySet()){
            double emp = precos.get(key);
            precos.put(key, emp + emp*factor);
        }
        System.out.println("Novo Catálogo:");
        for(String key:precos.keySet()){
            System.out.printf(" - %s: R$%.2f\n", key, precos.get(key));
        }
    }

    public static void main(String[] args) {
        System.out.println("Automação de Convites (Setor de Eventos/RH)");
        Q1();
        System.out.println("Cálculo de Comissão Progressiva (Setor de Vendas)");
        Q2();
        System.out.println("Verificação de Estoque Crítico (Setor de Logística)");
        Q3();
        System.out.println("Análise de Custos Mensais (Setor Financeiro)");
        Q4();
        System.out.println("Reajuste Geral de Preços (Setor Comercial)");
        Q5();
    }
}

File: My Answer/E7.cpp
#include <iostream>
#include <string>
#include <vector>
#include <stdio.h>
#include <stdlib.h>
#include <omp.h>
#include <iomanip>
#include <algorithm>
#include <cctype>
#include <locale>
#include <regex>
#include <cmath>
#include <stdexcept>
#include "utils.h"

using namespace std;

void Q1(){
    for(int i = 10; i >= 0; i--){
        cout << "O treinamento de Python (C++) começa em " << i << " minutos.\n";
    }
}

void Q2(){
    vector<float> vendas = {2000, 5000, 1000, 8000, 3000};
    int i = 1;
    for(float venda: vendas){
        bool temp = (venda <= 4000);
        cout << "Comissão da venda " << i << ": " << (temp ? "5%" : "10%") << endl;
        cout << "Valor da comissão: R$" << fixed << setprecision(2) << (temp ? venda*0.05 : venda*0.10) << endl;
        i++;
    }
}

void Q3(){
    vector<string> estoque_produtos = {"monitor", "teclado", "mouse", "headset", "gabinete"};
    vector<int> estoque_quantidades = {5, 12, 2, 8, 15};
    for(int i = 0; i < estoque_quantidades.size(); i++){
        if(estoque_quantidades[i] < 8) cout << "Alerta! O produto " << estoque_produtos[i] << " está com apenas " << estoque_quantidades[i] << " unidades no estoque!" << endl;
    }
}

void Q4(){
    map<string, float> metas = {
        {"jan", 1000},
        {"fev", 1200},
        {"mar", 1100}
    };
    map<string, float> gastos = {
        {"jan", 900},
        {"fev", 1350},
        {"mar", 1100}
    };
    for(const auto& [meta, valor] : metas){
        if(valor >= gastos[meta]) cout << "Mes " << meta << ": Dentro do orçamento." << endl;
        else cout << "Mes " << meta << ": Orçamento estourado em: " << fixed << setprecision(2) << (gastos[meta] - valor) << endl;
    }
}
void Q5(){
    map<string, float> precos = {
        {"celular", 1500},
        {"tablet", 2500},
        {"notebook", 5000}
    };
    cout<<"Digite o percentual de aumento: ";
    float percentual;
    cin >> percentual;
    cout<<"Novo Catálogo:"<<endl;
    for(const auto& [produto, valor] : precos){
        precos[produto] = valor*(1 + percentual/100);
        cout << " - " << produto << ": R$" << fixed << setprecision(2) << precos[produto] << endl;
    }
}

int main(){
    Q1();
    cout<<endl;
    Q2();
    cout<<endl;
    Q3();
    cout<<endl;
    Q4();
    cout<<endl;
    Q5();
    return 0;
}